Dockerfile

```diff
@@ -2,63 +2,42 @@ FROM alpine:3.7
 
 MAINTAINER Andy Driver <andy.driver@digital.justice.gov.uk>
 
-# install build dependencies (they'll be uninstalled after pip install)
-RUN apk add --no-cache \
-        --virtual build-deps \
-        gcc \
-        musl-dev
+ENV HELM_VERSION 2.9.1
+ENV HELM_HOME /tmp/helm
+ENV DJANGO_SETTINGS_MODULE "control_panel_api.settings"
+ENV USE_VENV=false
 
-# install python3 and 'ca-certificates' so that HTTPS works consistently
+WORKDIR /home/control-panel
+
+# install build dependencies (they'll be uninstalled after pip install)
 RUN apk add --no-cache \
+        build-base \
         openssl \
         ca-certificates \
         libffi-dev \
-        python3-dev
-
-# Temporary bugfix for libressl
-# Postgres needs libressl-dev, but cryptography only works with openssl-dev
-RUN apk add --no-cache --virtual temp-ssl-fix \
-        openssl-dev \
-    && pip3 install cryptography==2.2.2 \
-    && apk del temp-ssl-fix \
-    && apk add --no-cache \
+        python3-dev \
         libressl-dev \
         postgresql-dev
 
-# Install helm
-ENV HELM_VERSION 2.9.1
-RUN wget https://storage.googleapis.com/kubernetes-helm/helm-v$HELM_VERSION-linux-amd64.tar.gz \
-    && tar xzf helm-v$HELM_VERSION-linux-amd64.tar.gz \
-    && mv linux-amd64/helm /usr/local/bin \
-    && rm -rf helm-v$HELM_VERSION-linux-amd64.tar.gz linux-amd64
-
-# Configure helm
-ENV HELM_HOME /tmp/helm
-RUN helm init --client-only
+# Install and configure helm
 COPY helm-repositories.yaml /tmp/helm/repository/repositories.yaml
-RUN helm repo update
-
-WORKDIR /home/control-panel
+COPY Makefile ./
+RUN make install-helm \
+ && helm init --client-only \
+ && helm repo update
 
-# install python dependencies
-ADD requirements.txt requirements.txt
-RUN pip3 install -r requirements.txt
+# install python dependencies (and then remove build dependencies)
+COPY requirements.txt ./
+RUN make dependencies \
+ && apk del build-base \
+ && apk add --no-cache make
 
-# uninstall build dependencies
-RUN apk del build-deps
-
-ENV DJANGO_SETTINGS_MODULE "control_panel_api.settings"
-
-ADD manage.py manage.py
-ADD run_api run_api
-ADD run_tests run_tests
-ADD wait_for_db wait_for_db
-ADD control_panel_api control_panel_api
-ADD moj_analytics moj_analytics
+COPY control_panel_api control_panel_api
+COPY moj_analytics moj_analytics
+COPY manage.py wait_for_db ./
```

Comment:
Is this `wait_for_db` still used?

Reply:
if you're using docker-compose it's useful

Reply:
Right. I was following the `Makefile` dependencies 🎢  and the `wait_for_db` task is a dependency of `migrations` which is run by `docker-compose` 😕  👍

Reply:
Yeah, I eventually found from where this script was invoked 👍

Reply:
I removed `make` from the docker image, so the `docker-compose.yml` now runs `wait_for_db` directly.